cli/cli.go

```diff
@@ -577,10 +579,27 @@ func (runOptions *runOptionsType) handleLogFlags(ctx *cli.Context) error {
 	}
 	if ctx.IsSet("no-syslog") &&
 		!runOptions.logOptions.noSyslog {
-		if err := log.AddSyslogHook(); err != nil {
+		var err error
+		// NOTE: Always logging at the debug level to the syslog
+		//
+		// ==============================================================================
+		// REVIEWER-NOTE: This was changed slightly from the existing functionality in:
+		// mendersoftware/log addSysLogHook. The code there also had a weird comment
+		// about setting up the syslog hook which I did not understand:
+		//
+		// comment:
+		// Add the syslog hook to the logger. This is better than adding it directly,
+		// for the reasons described in the loggingHookType comments.
+		//
+		// ==============================================================================
+		hook, err := logrus_syslog.NewSyslogHook("", "",
+			syslog.LOG_DEBUG, "mender")
```

Comment:
Why can't you reuse the global log level? I think logging at debug level is not a good idea, because this will flood device logs.

Reply:
Yes, that's what I would think also. But this is how it is now. I thought it weird also, so figured I'd see if anyone knew the reason. 

Anywho, since u also think it weird, changing it. 

This was also reported on MH I think. I think I even created a ticket for it.

Here it is: https://tracker.mender.io/browse/MEN-3135

Great opportunity for two birds with one stone here :raised_back_of_hand:

Reply:
Okay, so I've looked into this a bit closer, and I still can't figure out how this is supposedly wrong.

So, my findings:

The `syslog.LOG_DEBUG`, is the priority given to the syslog message, with the std, `LOG_KERNEL` as the facility into which it logs.

But, this does not mean that this controls what is written to the syslog, only the severity given to it. In fact, the syslog hook in `logrus` will log all messages, no matter what is set as `log.Level`, or what the `syslog.Level` is here in the line above.

So this brings the question. Do we want to change this?

Reply:
This is the `logrus.syslog` hook:
```
func (hook *SyslogHook) Levels() []logrus.Level {
	return logrus.AllLevels
}
```
And this is the go syslog doc:
https://godoc.org/log/syslog#Priority

Reply:
Good investigation! Ideally, yes, we should change it, but I think you should keep that work out of this pull request, since it's not really related; this has always been like this, right?